### PR TITLE
Add FHIR parsing support

### LINKED ISSUE #102: Add reading support for further common EHR file formats
- [ ] fhir (https://www.hl7.org/fhir/overview.html) -> make use of https://github.com/fhirpack/main ?

### PR DESCRIPTION
**Pull request by Zethson**
Fixes #102 

1. Adds FHIR parsing support (json & ndjson)
2. Moves some of the original parsing code around to better group the file types (csv, h5ad)

### CODE REVIEW

File: ehrapy/io/_read.py [hunk 1]
```diff
@@ -359,6 +351,98 @@ def _do_read_h5ad(filename: Path | Iterator[str]) -> AnnData:
...
+        raise ValueError("Only folders containing json and ndjson in FHIR format are supported.")
+
+    df, columns_obs_only = _prepare_dataframe(df, columns_obs_only, columns_x_only, cache)
+    if index_column:
+        df.set_index(index_column)
+
+    if return_df:
+        return df
+    else:
+        adata = df_to_anndata(df, columns_obs_only)
```
Comment (Imipenem) [inline ehrapy/io/_read.py:436]: don't we need to use "columns_x_only" here as well?

File: tests/io/test_read.py [hunk 2]
```diff
@@ -262,3 +256,9 @@ def test_read_multiple_csv_with_x_only_2(self):
         assert set(adatas["dataset_num_with_missing"].obs.columns) == {"col2"}
         assert set(adatas["dataset_non_num_with_missing"].var_names) == {"strcol", "intcol", "boolcol"}
         assert set(adatas["dataset_num_with_missing"].var_names) == {"col1", "col3"}
+
```
Comment (Imipenem) [inline tests/io/test_read.py:259]: I'd like one or more additional tests, for example with "columns_obs_only" etc. being used

**Review by Imipenem — CHANGES_REQUESTED** [review]
Minor things, otherwise LGTM.

File: ehrapy/io/_read.py [hunk 3]
```diff
@@ -159,17 +127,41 @@ def _read_csv(
...
+    """
+    file: Path = Path(dataset_path)
+    if not file.exists():
+        file = _get_non_existing_files(file, download_dataset_name, backup_url)
 
-    # If the filename is a directory, assume it is a dataset with h5ad multiple files
-    if filename.is_dir():
-        return _read_from_directory(filename, False, None, "h5ad")
-    # dataset is a single h5ad file
+    if file.is_dir():
```
[the remark this answers is not in the document]
Comment (Zethson) [reply ehrapy/io/_read.py:159]: Renamed to `file_path`

File: ehrapy/io/_read.py (hunk 1 above)
Comment (Zethson) [reply ehrapy/io/_read.py:436]: No. The function can't take this parameter (yet). Any reason why?

**Pull request merged**